shell/shared/grain.js

```diff
@@ -511,6 +511,62 @@ if (Meteor.isClient) {
             currentGrainId + event.data.setPath);
       } else if (event.data.setTitle) {
         Session.set("grainFrameTitle", event.data.setTitle);
+      } else if (event.data.renderTemplate) {
+        // Request creation of a single-use template with a privileged API token.
+        // Why?  Apps should not be able to obtain capabilities-as-keys to
+        // themselves directly, because those can be leaked through an
+        // arbitrary bit stream or covert channel.  However, apps often need a
+        // way to provide instructions to users to copy/paste with some
+        // privileged token contained within.  By providing this templating in
+        // the platform, we can ensure that the token is only visible to the
+        // shell's origin.
+        var call = event.data.renderTemplate;
+        check(call, Object);
+        var rpcId = call.rpcId;
+        check(rpcId, String);
+        var template = call.template;
+        check(template, String);
+        var assignment = undefined;
+        if (call.roleId) {
```

Comment:
Does this mean that these templates can't be used for `allAccess` tokens? Is there any reason to allow an arbitrary Grain.ViewSharingLink.RoleAssignment here?